### ISSUE TITLE
Hello, I have completed the basic reproduction. But could you please tell me how to apply it to my own dataset? My dataset for splitting only has labels of 0 and 1. How should I proceed with this? Please help me. Thank you very much!

### ISSUE THREAD

**Issue by liubaoning111** [LanqingL/SCS#7]

**Comment by LanqingL** [comment]
Thank you for your attention to our work. Here are some key implementation points:

## Step 1: Create Dataset Class File

### Create `data/your_dataset.py`

**Requirements:**
- Implement a class that inherits from `torch.utils.data.Dataset`
- Implement required methods: `__init__`, `__len__`, `__getitem__`, `sample_episode`, etc.
- Return standardized batch dictionary (containing `query_img`, `query_mask`, `support_imgs`, etc.)

**Key Implementation Points:**
- `__init__` method must accept parameters compatible with `dataset.py`: `datapath`, `fold`, `transform`, `split`, `shot`, `use_original_imgsize`
- `__getitem__` method must return a batch dictionary with all required keys
- Implement helper methods: `_load_frames()`, `_read_img()`, `_read_mask()`, `_extract_ignore_idx()`

**Reference files:**
(https://github.com/user-attachments/files/23290170/iSALD_5i.py)

---

## Step 2: Prepare Data Files and Split Files

### 2.1 Dataset Directory Structure
Create the dataset directory with the following structure:
```
your_dataset/
├── train/
│   ├── images/          # Training images
│   └── masks/           # Training masks
└── val/
    ├── images/          # Validation images
    └── masks/           # Validation masks
```

### 2.2 Split Files
Create split files in `data/splits/your_dataset/`

**Supported formats:**
- **JSON format** (like ottawa, aeroscapes, coco): `train.json`, `val.json`
- **TXT format** (like pascal): `fold0.txt`, `fold1.txt`, etc.
- **Dictionary format** (like iSAID): `train_dict.txt`, `val_dict.txt`

**Notes:**
- Ensure mask encoding matches your dataset format (class_id + 1 or binary)
- Image and mask naming conventions must be consistent

---

## Step 3: Register in `dataset.py`

### 3.1 Import the New Dataset Class
Add import statement at the top of `data/dataset.py`:
```python
from your_dataset import YourDataset
```

### 3.2 Register in `cls.datasets` Dictionary
Add entry in `FSSDataset.initialize()` method:
```python
cls.datasets = {
    'pascal': DatasetPASCAL,
    # ... other datasets
    'your_dataset': YourDataset,  # Add this line
}
```

### 3.3 Add Special Handler (if needed)
If your dataset requires special handling (like iSAID), add a condition in `build_dataloader()`:
```python
@classmethod
def build_dataloader(cls, benchmark, bsz, nworker, fold, split, shot=1):
    if benchmark == 'your_dataset':
        return cls.build_your_dataset_dataloader(benchmark, bsz, nworker, fold, split, shot=shot)
    # ... other conditions
```

Then implement the custom dataloader builder method:
```python
@classmethod
def build_your_dataset_dataloader(cls, benchmark, bsz, nworker, fold, split, shot=1):
    shuffle = split == 'trn'
    nworker = nworker if split == 'trn' else 0
    transform = cls.trn_transform if split == 'trn' else cls.transform
    if split == 'trn':
        split = 'train'
    
    dataset = cls.datasets[benchmark](
        datapath=cls.datapath, 
        fold=fold, 
        transform=transform,
        shot=shot, 
        mode=split, 
        use_original_imgsize=cls.use_original_imgsize
    )
    dataloader = DataLoader(dataset, batch_size=bsz, shuffle=shuffle, 
                           num_workers=nworker, drop_last=True)
    return dataloader
```

---

## Key Points

### Batch Dictionary Format
The `__getitem__` method must return a batch dictionary with the following structure:
```python
batch = {
    'query_img': torch.Tensor,           # Query image (transformed)
    'query_mask': torch.Tensor,         # Query mask (binary or class mask)
    'query_name': str,                   # Query image name/path
    'query_ignore_idx': torch.Tensor,    # Ignore region mask (optional)
    'org_query_imsize': tuple,           # Original image size (width, height)
    
    'support_imgs': torch.Tensor,        # Support images (stacked tensor, shape: [shot, C, H, W])
    'support_masks': list,               # Support masks list (each element is a tensor)
    'support_names': list,               # Support image names list
    'support_ignore_idxs': list,         # Support ignore regions list (optional)
    
    'class_id': torch.Tensor             # Current sample class ID
}
```

### Important Notes
- **Keep interface consistency** with existing datasets
- **Ensure mask encoding** matches your format (class_id + 1 vs binary)
- **Avoid hardcoded paths** - use relative or configurable paths
- **Episode sampling logic** must ensure query image is not in support set

---

## Checklist

- [ ] Create `data/your_dataset.py` with complete dataset class implementation
- [ ] Prepare dataset directory structure (images/masks)
- [ ] Create split files in `data/splits/your_dataset/`
- [ ] Import new dataset class in `data/dataset.py`
- [ ] Register dataset in `cls.datasets` dictionary
- [ ] Add custom dataloader builder if needed
- [ ] Test data loading and verify batch format
- [ ] Verify episode sampling logic

**Comment by liubaoning111** [comment]
Thank you for your answer.If it's just a single dataset, can I try to set it up according to the "dataset" provided in the PASCAL-5i data set? Because I found that all I needed to do was to design the "flod.txt" properly.

**Comment by Linzeyin** [comment]
Are you sure you can reproduce the **data from the paper**? Are the hyperparameters the default ones used in the code?




> No description provided.
